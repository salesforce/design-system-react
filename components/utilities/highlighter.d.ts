declare module '@salesforce/design-system-react/components/utilities/highlighter' {
	import React from 'react';
	type Props = {
		/**
		 * The full string to display.
		 */
		children?: string | number | boolean | React.ReactNode;
		className?: string;
		/**
		 * The string of text (or Regular Expression) to highlight.
		 */
		search?: PropTypes.any;
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
