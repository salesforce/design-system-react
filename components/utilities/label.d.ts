declare module '@salesforce/design-system-react/components/utilities/label' {
	import React from 'react';
	type Props = {
		/*
		 * Assistive Text to use instead of a visible label
		 */
		assistiveText?: Record<string, any>;
		/**
		 * Class names to be added to the label
		 */
		className?: any[] | Record<string, any> | string;
		/*
		 * Id of the input associated with this label
		 */
		htmlFor?: string;
		/*
		 * Input Label
		 */
		label?: string;
		/*
		 * Applies label styling for a required form element
		 */
		required?: boolean;
		/**
		 * Changes markup of label.
		 */
		variant?: 'base' | 'static';
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
