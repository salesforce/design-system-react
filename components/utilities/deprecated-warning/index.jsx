/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';

// ## Constants
import { DEPRECATED_WARNING } from '../../../utilities/constants';

import Alert from '../../alert';
import AlertContainer from '../../alert/container';

/**
 * A utility component that is used to highlight a deprecated component.
 */
const DeprecatedWarning = () => (
	<div className="slds-p-top_x-large">
		<AlertContainer>
			<Alert
				labels={{
					heading: (
						<span>
							<strong>Deprecated component.</strong> No fit and finish bug fixes
							will be accepted for this component.
						</span>
					),
				}}
				variant="error"
			/>
		</AlertContainer>
	</div>
);

// ### Display Name
DeprecatedWarning.displayName = DEPRECATED_WARNING;

export default DeprecatedWarning;
