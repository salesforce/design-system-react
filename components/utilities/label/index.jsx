/* eslint-disable react/jsx-curly-brace-presence */
/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const propTypes = {
	/*
	 * Assistive Text to use instead of a visible label
	 */
	assistiveText: PropTypes.object,
	/**
	 * Class names to be added to the label
	 */
	className: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.object,
		PropTypes.string,
	]),
	/*
	 * Id of the input associated with this label
	 */
	htmlFor: PropTypes.string,
	/*
	 * Input Label
	 */
	label: PropTypes.string,
	/*
	 * Applies label styling for a required form element
	 */
	required: PropTypes.bool,
	/**
	 * Changes markup of label.
	 */
	variant: PropTypes.oneOf(['base', 'static']),
};

/*
 * Form label. This returns null if there is no label text (hidden or shown)
 */
const Label = ({
	variant = 'base',
	label,
	assistiveText,
	className,
	htmlFor,
	required,
}) => {
	const labelText = label || (assistiveText && assistiveText.label); // One of these is required to pass accessibility tests

	const subRenders = {
		base: (
			<label
				className={classNames(
					'slds-form-element__label',
					{
						'slds-assistive-text': assistiveText && !label,
					},
					className
				)}
				htmlFor={htmlFor}
			>
				{required && (
					<abbr className="slds-required" title="required">
						{'*'}
					</abbr>
				)}
				{labelText}
			</label>
		),
		static: (
			<span className={classNames('slds-form-element__label', className)}>
				{labelText}
			</span>
		),
	};

	return labelText ? subRenders[variant] : null;
};

Label.displayName = 'Label';
Label.propTypes = propTypes;

export default Label;
