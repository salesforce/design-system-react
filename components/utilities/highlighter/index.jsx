/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';
import PropTypes from 'prop-types';

// ### ReactHighlighter
import ReactHighlighter from 'react-highlighter';

// ## Constants
import { HIGHLIGHTER } from '../../../utilities/constants';

/**
 * A utility component that highlights occurrences of a particular pattern in its contents.
 */
const Highlighter = (props) => {
	if (props.search) {
		let children;
		if (typeof props.children === 'string') {
			children = (
				<ReactHighlighter
					className={props.className}
					matchClass={null}
					matchElement="mark"
					search={props.search}
					title={props.children}
				>
					{props.children}
				</ReactHighlighter>
			);
		} else {
			const findString = (nodeArr) =>
				nodeArr.map((element) => {
					let newElement;
					if (typeof element === 'string') {
						newElement = (
							<ReactHighlighter
								key={element}
								className={props.className}
								matchClass={null}
								matchElement="mark"
								search={props.search}
								title={element}
							>
								{element}
							</ReactHighlighter>
						);
					} else {
						newElement = element;
					}
					return newElement;
				});

			if (props.children.props) {
				const node = props.children.props.children;
				children = node instanceof Array ? findString(node) : node;
			}
		}

		return <span>{children}</span>;
	}

	if (typeof props.children === 'string') {
		return (
			<span className={props.className} title={props.children}>
				{props.children}
			</span>
		);
	}

	return <span className={props.className}>{props.children}</span>;
};

// ### Display Name
Highlighter.displayName = HIGHLIGHTER;

// ### Prop Types
Highlighter.propTypes = {
	/**
	 * The full string to display.
	 */
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.bool,
		PropTypes.node,
	]),
	className: PropTypes.string,
	/**
	 * The string of text (or Regular Expression) to highlight.
	 */
	search: PropTypes.any,
};

export default Highlighter;
