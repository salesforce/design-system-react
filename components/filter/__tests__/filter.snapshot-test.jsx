import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	testDOMandHTML,
	testImageSnapshot
} from '../../../tests/snapshot-helpers';

import Default from '../__examples__/default';
import NewFilter from '../__examples__/new';
import LockedFilter from '../__examples__/locked';
import PermanantFilter from '../__examples__/permanant';
import ErrorFilter from '../__examples__/error';
import AssistiveTextFilter from '../__examples__/assistive-text';

import { FILTER } from '../../../utilities/constants';

describe(FILTER, () => {
	test('Default DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(Default);
	});

	test('Default with custom className DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(Default, { className: 'MY_CUSTOM_CLASS_NAME' });
	});

	test('NewFilter DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(NewFilter);
	});

	test('LockedFilter DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(LockedFilter);
	});

	test('PermanantFilter DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(PermanantFilter);
	});

	test('ErrorFilter DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(ErrorFilter);
	});

	test('AssistiveTextFilter DOM & HTML Snapshots look the same', () => {
		testDOMandHTML(AssistiveTextFilter);
	});

	test('Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'Filter');
	});

	test('New Filter Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'New Filter');
	});

	test('Locked Filter Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'Locked Filter');
	});

	test('Permanant Filter Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'Permanant Filter');
	});

	test('Align Right Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'Filter Align Right');
	});

	test('AssistiveTextFilter Image Snapshot looks the same', async () => {
		await testImageSnapshot(FILTER, 'AssistiveTextFilter');
	});
});
