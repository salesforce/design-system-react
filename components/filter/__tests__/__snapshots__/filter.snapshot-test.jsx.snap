// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssistiveText Filter 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "editFilter-TEST",
        "editFilterHeading": "editFilterHeading-TEST",
        "removeFilter": "removeFilter-TEST",
      }
    }
    id="sample-panel-filtering-show-me"
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <OnClickOutside(SLDSCombobox)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      events={
        Object {
          "onSelect": [Function],
        }
      }
      excludeScrollbar={false}
      labels={
        Object {
          "label": "Show Me",
          "placeholder": "Select record type",
        }
      }
      menuPosition="relative"
      options={
        Array [
          Object {
            "id": 1,
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "id": 2,
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      selection={
        Array [
          Object {
            "id": 1,
            "label": "All Products",
            "value": "all-products",
          },
        ]
      }
      stopPropagation={false}
      variant="readonly"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`AssistiveText Filter HTML Snapshot 1`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`Error Filter Base Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "Edit filter:",
        "editFilterHeading": "Choose filter criteria",
      }
    }
    id="sample-panel-filtering-show-me"
    isError={true}
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <SLDSMenuPicklist
      checkmark={true}
      inheritTargetWidth={true}
      isInline={true}
      label="Show Me"
      labels={
        Object {
          "multipleOptionsSelected": "Multiple Options Selected",
        }
      }
      menuPosition="absolute"
      onSelect={[Function]}
      options={
        Array [
          Object {
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      placeholder="Select record type"
      value="all-products"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`Filter Base Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "Edit filter:",
        "editFilterHeading": "Choose filter criteria",
      }
    }
    id="sample-panel-filtering-show-me"
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <OnClickOutside(SLDSCombobox)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      events={
        Object {
          "onSelect": [Function],
        }
      }
      excludeScrollbar={false}
      labels={
        Object {
          "label": "Show Me",
          "placeholder": "Select record type",
        }
      }
      menuPosition="relative"
      options={
        Array [
          Object {
            "id": 1,
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "id": 2,
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      selection={
        Array [
          Object {
            "id": 1,
            "label": "All Products",
            "value": "all-products",
          },
        ]
      }
      stopPropagation={false}
      variant="readonly"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`Filter Base with custom className Snapshot 1`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center MY_CUSTOM_CLASS_NAME\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`LockedFilter Base Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "Edit filter:",
        "editFilterHeading": "Choose filter criteria",
      }
    }
    id="sample-panel-filtering-show-me"
    isLocked={true}
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <SLDSMenuPicklist
      checkmark={true}
      inheritTargetWidth={true}
      isInline={true}
      label="Show Me"
      labels={
        Object {
          "multipleOptionsSelected": "Multiple Options Selected",
        }
      }
      menuPosition="absolute"
      onSelect={[Function]}
      options={
        Array [
          Object {
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      placeholder="Select record type"
      value="all-products"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`NewFilter Base Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "Edit filter:",
        "editFilterHeading": "Choose filter criteria",
      }
    }
    id="sample-panel-filtering-show-me"
    isNew={true}
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <SLDSMenuPicklist
      checkmark={true}
      inheritTargetWidth={true}
      isInline={true}
      label="Show Me"
      labels={
        Object {
          "multipleOptionsSelected": "Multiple Options Selected",
        }
      }
      menuPosition="absolute"
      onSelect={[Function]}
      options={
        Array [
          Object {
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      placeholder="Select record type"
      value="all-products"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`Permanant Filter Base Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSFilter
    align="left"
    assistiveText={
      Object {
        "editFilter": "Edit filter:",
        "editFilterHeading": "Choose filter criteria",
      }
    }
    id="sample-panel-filtering-show-me"
    isPermanent={true}
    onChange={[Function]}
    onRemove={[Function]}
    predicate="All Products"
    property="Show Me"
  >
    <SLDSMenuPicklist
      checkmark={true}
      inheritTargetWidth={true}
      isInline={true}
      label="Show Me"
      labels={
        Object {
          "multipleOptionsSelected": "Multiple Options Selected",
        }
      }
      menuPosition="absolute"
      onSelect={[Function]}
      options={
        Array [
          Object {
            "label": "All Products",
            "value": "all-products",
          },
          Object {
            "label": "All Wackamoles",
            "value": "all-Wackamoles",
          },
        ]
      }
      placeholder="Select record type"
      value="all-products"
    />
  </SLDSFilter>
</SLDSIconSettings>
`;

exports[`SLDSFilter AssistiveTextFilter DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby={undefined}
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        editFilter-TEST
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex={undefined}
    title="removeFilter-TEST"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--hint"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      removeFilter-TEST
    </span>
  </button>
</div>
`;

exports[`SLDSFilter AssistiveTextFilter DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">editFilter-TEST</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"removeFilter-TEST\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">removeFilter-TEST</span></button>
</div>"
`;

exports[`SLDSFilter Default DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby={undefined}
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        Edit filter:
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex={undefined}
    title="Remove Filter: Show Me All Products"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--hint"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Remove Filter: Show Me All Products
    </span>
  </button>
</div>
`;

exports[`SLDSFilter Default DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`SLDSFilter Default with custom className DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center MY_CUSTOM_CLASS_NAME"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby={undefined}
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        Edit filter:
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex={undefined}
    title="Remove Filter: Show Me All Products"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--hint"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Remove Filter: Show Me All Products
    </span>
  </button>
</div>
`;

exports[`SLDSFilter Default with custom className DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center MY_CUSTOM_CLASS_NAME\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`SLDSFilter ErrorFilter DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center slds-has-error"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby="sample-panel-filtering-show-me-error"
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        Edit filter:
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex={undefined}
    title="Remove Filter: Show Me All Products"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--hint"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Remove Filter: Show Me All Products
    </span>
  </button>
</div>
`;

exports[`SLDSFilter ErrorFilter DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center slds-has-error\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" aria-describedby=\\"sample-panel-filtering-show-me-error\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`SLDSFilter LockedFilter DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center slds-is-locked"
>
  <button
    aria-describedby={undefined}
    className="slds-button--reset slds-grow slds-has-blur-focus"
    disabled={true}
  >
    <p
      className="slds-text-body--small"
    >
      Show Me
    </p>
    <p>
      All Products
    </p>
  </button>
</div>
`;

exports[`SLDSFilter LockedFilter DOM & HTML Snapshots look the same 2`] = `"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center slds-is-locked\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" disabled=\\"\\"><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>"`;

exports[`SLDSFilter NewFilter DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center slds-is-new"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby={undefined}
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        Edit filter:
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
  <button
    aria-controls={undefined}
    aria-describedby={undefined}
    aria-expanded={undefined}
    aria-haspopup={undefined}
    className="slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small"
    disabled={false}
    id={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onFocus={undefined}
    onKeyDown={undefined}
    onKeyPress={undefined}
    onKeyUp={undefined}
    onMouseDown={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    tabIndex={undefined}
    title="Remove Filter: Show Me All Products"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="slds-button__icon slds-button__icon--hint"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Remove Filter: Show Me All Products
    </span>
  </button>
</div>
`;

exports[`SLDSFilter NewFilter DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center slds-is-new\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\\" title=\\"Remove Filter: Show Me All Products\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Show Me All Products</span></button>
</div>"
`;

exports[`SLDSFilter PermanantFilter DOM & HTML Snapshots look the same 1`] = `
<div
  className="slds-filters__item slds-grid slds-grid--vertical-align-center"
>
  <div
    className="slds-grow"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <button
      aria-describedby={undefined}
      className="slds-button--reset slds-grow slds-has-blur-focus"
      id="sample-panel-filtering-show-me"
      onClick={[Function]}
      onFocus={null}
      onMouseDown={undefined}
      onMouseEnter={null}
      onMouseLeave={null}
      tabIndex="0"
    >
      <span
        className="slds-assistive-text"
      >
        Edit filter:
      </span>
      <p
        className="slds-text-body--small"
      >
        Show Me
      </p>
      <p>
        All Products
      </p>
    </button>
  </div>
</div>
`;

exports[`SLDSFilter PermanantFilter DOM & HTML Snapshots look the same 2`] = `
"<div class=\\"slds-filters__item slds-grid slds-grid--vertical-align-center\\">
  <div class=\\"slds-grow\\" style=\\"display:inline;\\"><button class=\\"slds-button--reset slds-grow slds-has-blur-focus\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body--small\\">Show Me</p><p>All Products</p></button></div>
</div>"
`;
